### PR TITLE
Creating Enumerables method

### PR DESCRIPTION
**Pull request by jairjaramillo**

### CODE REVIEW

File: enum.rb [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      bubble_sort_by(arra) do |_left, _right|
+        yield(arra[x], arra[x + 1])
+      end
+    end
+    arra
+  end
+
+  def my_each(counter = 0)
+    while counter < to_a.length
+      yield to_a[counter]
```
Comment (samgaco) [inline enum.rb:37]: - [x] my_each returns an Enumerator if no block is given
- [x] my_each_with_index returns an enumerator if no block is given

File: enum.rb [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    while counter < to_a.length
+      yield to_a[counter], counter
+      counter += 1
+    end
+  end
+
+  def my_select
+    result = []
+    i = 0
+    while i < to_a.length
```
Comment (samgaco) [inline enum.rb:57]: - [x] my_select returns an enumerator if no block is given

File: enum.rb [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      result << to_a[i] if yield to_a[i]
+      i += 1
+    end
+    result
+  end
+
+  def my_all?(pattern = false)
+    if block_given?
+      my_each { |item| return false if !(yield item) == false }
+    elsif !pattern == true
```
Comment (samgaco) [inline enum.rb]: - [x] my_all? returns true if the block never returns false or nil
- [x] my_all? when no block or argument is given returns true when none of the collection members are false or nil
- [x] my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
- [x] my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [x] my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

File: enum.rb [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    else
+      my_each { |item| return false unless item == false }
+    end
+    true
+  end
+
+  def my_any?(pattern = false)
+    if block_given?
+      my_each { |item| return true if yield item }
+    elsif !pattern == true
```
Comment (samgaco) [inline enum.rb]: - [x] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- [x] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [x] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [x] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

File: enum.rb [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    else
+      my_each { |item| return true unless item }
+    end
+    false
+  end
+
+  def my_none?(pattern = false)
+    if block_given?
+      my_each { |item| return false if yield item }
+    elsif !pattern == true
```
Comment (samgaco) [inline enum.rb]: - [x] my_none? when no block or argument is given returns true only if none of the collection members is true
- [x] my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
- [x] my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
- [x] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

File: enum.rb [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    else
+      my_each { |item| return false unless item }
+    end
+    true
+  end
+
+  def my_count(pattern = false)
+    count = 0
+    if block_given?
+      my_each { |item| count += 1 if yield item }
```
Comment (samgaco) [inline enum.rb]: - [x] my_count returns the number of items in enum through enumeration
- [x] my_count counts the number of items in enum that are equal to item if an argument is given

File: enum.rb [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+      my_each { |item| count += 1 if pattern == item }
+    else
+      my_each { |item| count += 1 unless item }
+    end
+    count
+  end
+
+  def my_map(proc = nil)
+    result = []
+    to_a.my_each do |item|
```
Comment (samgaco) [inline enum.rb:125]: - [x] my_map returns an Enumerator if no block is given

File: enum.rb [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+      elsif block_given?
+        result.push(yield item)
+      end
+    end
+    result
+  end
+
+  def my_inject(init = nil, &block)
+    return to_a[1..-1].my_inject(first, &block) if init.nil?
+
```
Comment (samgaco) [inline enum.rb:148]: - [x] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @jairjy :high_brightness: 

I left you some comments below about what functionalities your methods are lacking. Please contrast your methods against the original ones [here](https://ruby-doc.org/core-2.6.3/Enumerable.html).

If you would need further clarification with some concrete examples you may **ask me in the comments** below or slack, my name is _Samuel García_ over there.

* Don't forget to solve the conflicts in your branch as well.

This project is a bit tricky but it will help you with getting to know ruby better, so you will be feeling more fluent moving on, therefore good spirit :muscle: 

-------------------------------------

* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

File: enum.rb [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+      my_each { |x| return false unless x.class == pattern }
+    elsif pattern.class == Regexp
+      my_each { |x| return false unless x =~ pattern }
+    elsif pattern.nil?
+      my_each { |x| return false unless x }
+    else
+      my_each { |x| return false unless x == pattern }
+    end
+    true
+  end
```
Comment (stratospherique) [inline enum.rb]: * [ ] my_all? when no block or argument is given returns true when none of the collection members are false or nil

File: enum.rb [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+      my_each { |x| return true if x.class == pattern }
+    elsif pattern.class == Regexp
+      my_each { |x| return true if x =~ pattern }
+    elsif pattern.nil?
+      my_each { |x| return true if x }
+    else
+      my_each { |x| return true if x == pattern }
+    end
+    false
+  end
```
Comment (stratospherique) [inline enum.rb]: * [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil

File: enum.rb [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+      my_each { |x| return false if x.class == pattern }
+    elsif pattern.class == Regexp
+      my_each { |x| return false if x =~ pattern }
+    elsif pattern.nil?
+      my_each { |x| return false if x }
+    else
+      my_each { |x| return false if x == pattern }
+    end
+    true
+  end
```
Comment (stratospherique) [inline enum.rb]: * [ ] my_none? when no block or argument is given returns true only if none of the collection members is true

File: enum.rb [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+    counter = 0
+    if block_given?
+      my_each { |x| counter += 1 if yield(x) == true }
+    elsif things.nil?
+      my_each { counter += 1 }
+    else
+      my_each { |x| counter += 1 if x == things }
+    end
+    counter
+  end
```
Comment (stratospherique) [inline enum.rb]: * [ ] my_count returns the number of items in enum through enumeration

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work implementing the suggested changes :clap: . However, some methods still don't meet with the requirements. Check the code comments below for the requirements :arrow_down: 

- **[REQUIRED]** fix the merging conflict that you have in this branch

### That's it :relaxed: . When finished, kindly resubmit another code review request :pray: 

_happy coding_ :v: 
**Ahmed Mahfoudh**

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Hi jairjy

Everything looks good with your enumerable module. :100: 

However, you have a  conflict to resolve: 

```
      This branch has conflicts that must be resolved
       Conflicting files
       .rubocop.yml
 ```
You will find this information :point_up_2: at the buttom of this pull request.


 **After resolving the conflict, submit another form for a code review**

#### ``` Have a great day ``` :slightly_smiling_face: 

@ Uchechukwu Henry Inyama

**Review by uche-inyama — APPROVED** [review]
## Hi jairjy!

Everything is good now, go ahead and merge. :clap: 

#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```

**Pull request merged**